program/src/matching.rs

```diff
@@ -1428,40 +1354,69 @@ impl<'a> Book<'a> {
         let now_ts = Clock::get()?.unix_timestamp as u64;
         let max_depth: i64 = perp_market.liquidity_mining_info.max_depth_bps.to_num();
 
+        let market_index_u8 = market_index as u8;
         let mut bids_keys = vec![];
         let mut asks_keys = vec![];
         for i in 0..MAX_PERP_OPEN_ORDERS {
-            if mango_account.order_market[i] != market_index as u8 {
+            if mango_account.order_market[i] != market_index_u8 {
                 continue;
             }
             match mango_account.order_side[i] {
                 Side::Bid => bids_keys.push(mango_account.orders[i]),
                 Side::Ask => asks_keys.push(mango_account.orders[i]),
             }
         }
-        bids_keys.sort_unstable();
-        asks_keys.sort_by(|a, b| b.cmp(a));
+        self.cancel_all_bids_with_size_incentives(
+            mango_account,
+            perp_market,
+            market_index,
+            max_depth,
+            now_ts,
+            &mut limit,
+            bids_keys,
+        )?;
+        self.cancel_all_asks_with_size_incentives(
+            mango_account,
+            perp_market,
+            market_index,
+            max_depth,
+            now_ts,
+            &mut limit,
+            asks_keys,
+        )
+    }
 
-        // First do bids
+    /// Internal
+    fn cancel_all_bids_with_size_incentives(
```

Comment:
It sounds like `cancel_all_bids_with_size_incentives` and `cancel_all_asks_with_size_incentives` are nearly the same function. Could they be merged and take the side as an arg?

Reply:
They probably could, but then I'd have to switch up greater than or less than symbols everywhere based on side. Going to be a bit ugly either way.

Reply:
Yeah, perfectly ok to not do in this commit, just a general observation that it'd be easy for bid/ask code to get subtly out of sync right now